lib/containers/container_images.pm

```diff
@@ -94,40 +102,40 @@ sub test_opensuse_based_image {
     my $version = $args{version} //= get_required_var("VERSION");
 
     die 'Argument $image not provided!'   unless $image;
     die 'Argument $runtime not provided!' unless $runtime;
 
     $version = 'Tumbleweed' if ($version =~ /^Staging:/);
 
-    # It is the right version
-    if ($distri eq 'sle') {
-        my $pretty_version = $version =~ s/-SP/ SP/r;
-        record_info "Validating", "Validating That $image has $pretty_version on /etc/os-release";
-        validate_script_output("$runtime container run --entrypoint '/bin/bash' --rm $image -c 'grep PRETTY_NAME /etc/os-release' | cut -d= -f2",
-            sub { /"SUSE Linux Enterprise Server $pretty_version"/ });
+    $version =~ s/^Jump://i;
+    validate_script_output qq{$runtime container run --entrypoint '/bin/bash' --rm $image -c 'cat /etc/os-release'}, sub { /PRETTY_NAME="openSUSE (Leap )?${version}.*"/ };
+    test_image_zyp_ref_and_verify($image, $runtime);
+}
+
+# Testing sle based images
+sub test_sle_based_image {
+    my %args    = @_;
+    my $image   = $args{image};
+    my $runtime = $args{runtime};
+
+    my $version = $args{version} //= get_required_var("VERSION");
 
+    die 'Argument $image not provided!'   unless $image;
+    die 'Argument $runtime not provided!' unless $runtime;
+
+    my $pretty_version = $version =~ s/-SP/ SP/r;
+    record_info "Validating", "Validating That $image has $pretty_version on /etc/os-release";
+    validate_script_output("$runtime container run --entrypoint '/bin/bash' --rm $image -c 'grep PRETTY_NAME /etc/os-release' | cut -d= -f2",
+        sub { /"SUSE Linux Enterprise Server $pretty_version"/ });
+    if (can_build_sle_base($image)) {
         my $plugin = '/usr/lib/zypp/plugins/services/container-suseconnect-zypp';
         assert_script_run "$runtime container run --entrypoint '/bin/bash' --rm $image -c '$plugin -v'";
         script_run "$runtime container run --entrypoint '/bin/bash' --rm $image -c '$plugin lp'", 420;
         script_run "$runtime container run --entrypoint '/bin/bash' --rm $image -c '$plugin lm'", 420;
-    } else {
-        $version =~ s/^Jump://i;
-        validate_script_output qq{$runtime container run --entrypoint '/bin/bash' --rm $image -c 'cat /etc/os-release'}, sub { /PRETTY_NAME="openSUSE (Leap )?${version}.*"/ };
+        test_image_zyp_ref_and_verify($image, $runtime);
     }
-
-    # zypper lr
-    assert_script_run("$runtime run --rm $image zypper lr -s", 120);
-    # zypper ref
-    assert_script_run("$runtime run --name refreshed $image sh -c 'zypper -v ref | grep \"All repositories have been refreshed\"'", 120);
-    # Commit the image
-    assert_script_run("$runtime commit refreshed refreshed-image", 120);
-    # Remove it
-    assert_script_run("$runtime rm refreshed", 120);
-    # Verify the image works
-    assert_script_run("$runtime run --rm refreshed-image sh -c 'zypper -v ref | grep \"All repositories have been refreshed\"'", 120);
```

Comment:
I'm happy for `test_image_zyp_ref_and_verify()` but it runs for openSUSE too.

Reply:
indeed. what s the problem?

Reply:
In your case it does not. In your case it just runs on SLE.

Reply:
I was wrong. I noticed it in `test_opensuse_based_image()` now too.